### ISSUE TITLE
Update the Schema Store `stylelintrc.json`

### ISSUE THREAD

**Issue by Mouvedia** [stylelint/stylelint#7973]
### What is the problem you're trying to solve?

This [JSON Schema file](https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/stylelintrc.json) is out of date.
e.g. `no-unknown-custom-media` and `declaration-property-value-no-unknown` are missing

### What solution would you like to see?

This file should be usable when retrieved using https://www.schemastore.org/

**Comment by ybiquitous** [comment]
@Mouvedia Thanks for opening the issue. But I think the suggestion is out of the scope of this repository since it's hosted in another repository.

Specifically, what do you want to achieve?

**Comment by Mouvedia** [comment]
> Specifically, what do you want to achieve?

This is a task.
This file is actually quite elaborate and it would be a shame if it wasn't kept up to date.
If you feel that it's not our responsibility you can close this issue.
We might even provide it ourselves alongside the d.ts files.

**Comment by ybiquitous** [comment]
I'm not sure about this `schemastore` repository, but how do other famous projects provide a config JSON schema? For example, `tsconfig.json` seems to be hosted in this repository, not in the TypeScript repository:
https://github.com/SchemaStore/schemastore/blob/7780fa1e1e48cbd5c13f998e8c24251689a73643/src/schemas/json/tsconfig.json

**Comment by Mouvedia** [comment]
The goal would be to link to it on https://stylelint.io/user-guide/configure once it has been updated.
e.g. prettier links to theirs on https://prettier.io/docs/en/configuration.html#configuration-schema

**Comment by ybiquitous** [comment]
How does Prettier the schema maintain?

**Comment by Mouvedia** [comment]
I love when you talk like Yoda :)

> How does Prettier the schema maintain?

No idea, but I am working on something related to schemata right now.
i.e. generate schema from config
So we should be able to patch it quite easily once I am done with my experiment.

I am fine with the task being postponed for now but I think it should be tracked.

**Comment by ybiquitous** [comment]
Sounds good. Maybe Prettier would also have a similar tool to auto-generate the JSON schema.

In this commit (https://github.com/SchemaStore/schemastore/commit/ed6af13eee1480dede796a67ea75ba37a57f85dd), it seems a Prettier's maintainer submitted a PR to the `schemastore` repo.

**Comment by Mouvedia** [comment]
SchemaStore/schemastore#4057

**Comment by Mouvedia** [comment]
SchemaStore/schemastore#4057

It's not perfect but it's an improvement.
I have a script that generates a 2020-12 JSON schema from the `possible` arguments which is more accurate and less cumbered by rule categories.

**Comment by hyperupcall** [comment]
> I'm not sure about this schemastore repository, but how do other famous projects provide a config JSON schema? For example, tsconfig.json seems to be hosted in this repository, not in the TypeScript repository:

Sorry if I'm butting in, but besides TypeScript and Prettier, we also get frequent PRs by maintainers of projects from projects like ruff (like [this](https://github.com/SchemaStore/schemastore/pull/4007) and [this](https://github.com/SchemaStore/schemastore/pull/3970)), from uv (like [this](https://github.com/SchemaStore/schemastore/pull/3919)), and [buf](https://github.com/SchemaStore/schemastore/pull/3883) (https://buf.build).

If there is a script that automatically generates the schema, would that be a useful thing to integrate into SchemaStore? I've experimented with this [a bit](https://github.com/fox-projects/jsonschema-extractor), but scripts written by a first-party probably work a lot better.

**Comment by Mouvedia** [comment]
@hyperupcall

The schema on schemastore.org has a different purpose.
It's meant to cover several major versions of Stylelint and support many tools.

The script I was talking about is brittle and has a purpose which exceed the scope of `stylelint/stylelint`.
i.e. 1. schema created from d.ts 2. schema created from d.ts from `@stylelint-types` (first fallback) 3. the script (second fallback)

tl;dr: even if we were to create a schema based on our `index.d.ts` it wouldn't include removed rules that were deprecated in the last major and it would either be 2020-12 JSON Schema or JSON Type Definition (i.e. not draft 07 compatible)

**Comment by hyperupcall** [comment]
@Mouvedia Ah I see, thank you for the clarification;
